src/mainwindow.cpp

```diff
@@ -33,9 +33,30 @@
 #include <QSortFilterProxyModel>
 
 #include "models/costmodel.h"
-#include "models/framedata.h"
 #include "parsers/perf/perfparser.h"
 #include "flamegraph.h"
+#include "models/summarydata.h"
+#include "models/topproxy.h"
+
+namespace {
+QString formatTimeString(quint64 nanoseconds)
+{
+    quint64 totalSeconds = nanoseconds / 1000000000;
+    quint64 days = totalSeconds / 60 / 60 / 24;
+    quint64 hours = (totalSeconds / 60 / 60) % 24;
+    quint64 minutes = (totalSeconds / 60) % 60;
+    quint64 seconds = totalSeconds % 60;
+    quint64 milliseconds = (nanoseconds / 1000000) % 1000;
+
+    auto format = [] (quint64 fragment, int precision = 2) -> QString {
+        return QString::number(fragment).rightJustified(precision, '0');
+    };
+    auto optional = [format] (quint64 fragment) -> QString {
+        return fragment > 0 ? format(fragment) + QLatin1Char(':') : QString();
+    };
+    return optional(days) + optional(hours) + optional(minutes) + format(seconds) + QLatin1Char('.') + format(milliseconds, 3);
```

Comment:
add a newline in here somewhere (in the code) to break this long line, i.e. something like

```
    return optional(days) + optional(hours) + optional(minutes)
           + format(seconds) + QLatin1Char('.') + format(milliseconds, 3);
```